pytensor/tensor/rewriting/math.py

```diff
@@ -423,6 +424,100 @@ def local_sumsqr2dot(fgraph, node):
                     return [new_out]
 
 
+@register_specialize
```

Comment:
Rethinking this, it may make sense to register these new rewrites in canonicalize as well.

Reply:
Hi, after looking around how it's resolved elsewhere and some experimentation with different options, I've ended up adding the following line at the beginning of the test function, which seems to have resolved the issue on my local runs in both with and without the FAST_COMPILE flag.

`    mode = get_mode("FAST_RUN") if config.mode == "FAST_COMPILE" else get_default_mode()
`

(Obviously, with passing the mode parameter to all function() calls)

If you're ok with this, on which branch do you want me to commit this change? The original one or the the one you had rebased that branch earlier?

Also, do you want me to add back the @register_canonicalize decorators?

Thanks!

Reply:
I think a more clean solution is to use `mode=get_default_mode.excluding("fusion")`

Then the tests will also be more straightforward because you don't need to check the graph inside the Composites

And yes, let's try to add the `register_canonicalize` to these rewrites and see if it doesn't break anything.

Reply:
An interesting side effect of making the rewrites canonical is that one of the test case in test_math.py::TestSigmoidRewrites::test_local_sigm_times_exp breaks, as now my rewrites takes precedence over the sigma_times_exp rewrites. More specifically,
`sigma(x) * sigma(-y) * (-exp(-x)) * exp(xy) * exp(y)`
is supposed to become
`(-1) * sigma(-x) * sigma(y) * exp(xy)`
but my rewrites contracts the exp-s before the sigma*exp rewrite could take action, and the result becomes
`(-1) * exp(xy+y-x) * sigma(x) * sigma(-y)`

So, I guess, this is a trade-off, how should we resolve this? (Having run all tests, this is the only one the canonicalisation broke.)

Reply:
We could maybe extend the sigma_exp rewrite to check if the balancing term is inside the exponentiation?

`exp(y - x) * sigmoid(x) -> exp(y) * sigmoid(x)`

Does that make sense? Might have other ramifications I am not seeing.

Reply:
That's a bit challenging looking at the rewrite so let's leave as is (don't canonicalize) and we can open an issue to investigate if it's worth it.

Reply:
I'm wondering how frequent / marginal such use cases are. Are there any kind of statistics available on how often real world use cases trigger which rewrites?

Meanwhile, I've just committed the changes for the tests, hope they'll be fine this time.

Thanks!

Reply:
Unfortunately, we have no telemetrics to assess types of graphs users create :D